blockchain/gen_genesis.go

```diff
@@ -60,6 +62,12 @@ func (g *Genesis) UnmarshalJSON(input []byte) error {
 		ParentHash *common.Hash                                `json:"parentHash"`
 	}
 	var dec Genesis
+
+	decoder := json.NewDecoder(bytes.NewReader(input))
+	decoder.DisallowUnknownFields()
+	if err := decoder.Decode(&dec); err != nil {
+		log.NewModuleLogger(log.CMDUtilsNodeCMD).Error("Unmarshal error", "error", err)
+	}
 	if err := json.Unmarshal(input, &dec); err != nil {
 		return err
 	}
```

Comment:
Do we still need this? `dec` is already unmarshal in `decoder.Decode(&dec)`.

Reply:
Only the "unknown field" is acceptable error in `Decode()`, but there was no specific error type for this. If the error is not the one, the following code should not be invoked and return early as the L72 does. 

Of course I could check like this `strings.Contains("unknown field")`, but it seemed ad-hoc. Any idea do you have to safely consolidate them?

Reply:
I thought something like `strings.Contains`, but it seems unnecessary. Could you add a little comment about this?

Reply:
Refactored. PTAL.